### PR TITLE
Jet id default refactoring

### PR DESCRIPTION
**Pull request by RHofsaess**
This PR introduces reasonable defaults for muonID and jetID.
The "to_lower" casting was moved to the related KappaEnumTypes::FUNCTION for the jetID stuff to reduce duplicated code.
Additionally, a logic to ensure a correct configuration was added to the ValidJetsProducerBase's passesJetID function.

### CODE REVIEW

File: KappaAnalysis/interface/Producers/ValidJetsProducer.h [hunk 1]
```diff
@@ -56,12 +56,9 @@ class ValidJetsProducerBase : public KappaProducerBase, public ValidPhysicsObjec
...
 
-        validJetsInput = KappaEnumTypes::ToValidJetsInput(
-                boost::algorithm::to_lower_copy(boost::algorithm::trim_copy(settings.GetValidJetsInput())));
-        jetIDVersion = KappaEnumTypes::ToJetIDVersion(
-                boost::algorithm::to_lower_copy(boost::algorithm::trim_copy(settings.GetJetIDVersion())));
-        jetID = KappaEnumTypes::ToJetID(
-                boost::algorithm::to_lower_copy(boost::algorithm::trim_copy(settings.GetJetID())));
+        validJetsInput = KappaEnumTypes::ToValidJetsInput(settings.GetValidJetsInput());
+        jetIDVersion = KappaEnumTypes::ToJetIDVersion(settings.GetJetIDVersion());
+        jetID = KappaEnumTypes::ToJetID(settings.GetJetID());
```
Comment (HerrHorizontal) [inline KappaAnalysis/interface/Producers/ValidJetsProducer.h:61]: Why did you remove the `to_lower`? I don't understand.

Comment (HerrHorizontal) [reply KappaAnalysis/interface/Producers/ValidJetsProducer.h:61]: Is it because it is already internally called in the according `KappaEnumTypes::` method?

File: KappaAnalysis/src/KappaEnumTypes.cc [hunk 2]
```diff
@@ -43,20 +44,26 @@ KappaEnumTypes::JetIDVersion KappaEnumTypes::ToJetIDVersion(std::string const& j
 	else if ((jetIDVersionLower == "2016ul") || (jetIDVersionLower == "ul2016")) return KappaEnumTypes::JetIDVersion::ID2016UL;
 	else if ((jetIDVersionLower == "2017ul") || (jetIDVersionLower == "ul2017")) return KappaEnumTypes::JetIDVersion::ID2017UL;
 	else if ((jetIDVersionLower == "2018ul") || (jetIDVersionLower == "ul2018")) return KappaEnumTypes::JetIDVersion::ID2018UL;
-	else LOG(FATAL) << "Jet ID version '" << jetIDVersion << "' is not available";
-	return KappaEnumTypes::JetIDVersion::ID2016;
+	else if (jetIDVersionLower == "none") return KappaEnumTypes::JetIDVersion::NONE;
+	else {
+	    LOG(FATAL) << "[KappaEnumTypes] Jet ID version '" << jetIDVersion << "' is not available";
+	    return KappaEnumTypes::JetIDVersion::NONE; // necessary but never happens!
```
Comment (HerrHorizontal) [inline KappaAnalysis/src/KappaEnumTypes.cc:50]: Isn't `LOG(FATAL)` terminating the program? This return shouldn't be needed.

File: KappaAnalysis/src/KappaEnumTypes.cc [hunk 3]
```diff
@@ -43,20 +44,26 @@ KappaEnumTypes::JetIDVersion KappaEnumTypes::ToJetIDVersion(std::string const& j
...
+	std::string jetIDLower = boost::algorithm::to_lower_copy(boost::algorithm::trim_copy(jetID));
+	if (jetIDLower == "loose") return KappaEnumTypes::JetID::LOOSE;
+	else if (jetIDLower == "looselepveto") return KappaEnumTypes::JetID::LOOSELEPVETO;
+	else if (jetIDLower == "medium") return KappaEnumTypes::JetID::MEDIUM;
+	else if (jetIDLower == "tight") return KappaEnumTypes::JetID::TIGHT;
+	else if (jetIDLower == "tightlepveto") return KappaEnumTypes::JetID::TIGHTLEPVETO;
+	else if (jetIDLower == "none") return KappaEnumTypes::JetID::NONE;
+	else {
+	    LOG(FATAL) << "Jet ID of type '" << jetID << "' not implemented!";
+	    return KappaEnumTypes::JetID::NONE; // necessary but never happens!
```
Comment (HerrHorizontal) [inline KappaAnalysis/src/KappaEnumTypes.cc:65]: See above.

**Review by HerrHorizontal — COMMENTED** [review]
I have some question left. See inline comments.

File: KappaAnalysis/interface/Producers/ValidJetsProducer.h (hunk 1 above)
Comment (RHofsaess) [reply KappaAnalysis/interface/Producers/ValidJetsProducer.h:61]: yes. This is moved in the KappaEnumTypes::function directly to ensure that the input is "to_lower", even if it is forgotten in the producer itself (+ in Excalibu'sr ZJetCutsFilter this is missing and with this we avoid problems and duplicate code)

File: KappaAnalysis/src/KappaEnumTypes.cc (hunk 2 above)
Comment (RHofsaess) [reply KappaAnalysis/src/KappaEnumTypes.cc:50]: It's a non-void function. Therefore, the compile does not allow a case without a return and our logging is not recognized as a return. So this (ugly) solution is necessary.

File: KappaAnalysis/src/KappaEnumTypes.cc (hunk 3 above)
Comment (RHofsaess) [reply KappaAnalysis/src/KappaEnumTypes.cc:65]: See above

**Review by HerrHorizontal — APPROVED** [review]
I have nothing to complain about. In my opinion, this is ready to merge.

**Review by cverstege — APPROVED** [review]
